sys/vm/vm_map.c

```diff
@@ -975,13 +991,24 @@ _vm_map_init(vm_map_t map, pmap_t pmap, vm_offset_t min, vm_offset_t max)
 	map->timestamp = 0;
 	map->busy = 0;
 	map->anon_loc = 0;
+#ifdef __CHERI_PURE_CAPABILITY__
+	/*
+	 * Do not enforce exact bounds here. The kernel map
+	 * can not be made representable without dropping some
+	 * physical memory, so restrict bounds as much as possible
+	 * and rely on the vm_map min/max enforcement.
+	 */
+	map->map_capability = cheri_setbounds((void *)min,
```

Comment:
I prefer casts on the result not the input (to make it clear it's a change of type, not because it's old code from before the builtins were overloaded. But should `map_capability` just be a `vm_pointer_t` instead? All the times it gets used it ultimately ends up being cast to that.

Reply:
Moved to use `vm_pointer_t` and cleared useless casts.

Reply:
Settled for exposing `vm_pointer_t` via sys/user.h